azurelinuxagent/ga/monitor.py

```diff
@@ -267,3 +256,84 @@ def add_sysinfo(self, event):
                                param.value)
                 event.parameters.remove(param)
         event.parameters.extend(self.sysinfo)
+
+    def send_host_plugin_heartbeat(self, protocol, last_host_plugin_heartbeat, host_plugin_errorstate, health_service):
+
+        """
+        Send a health signal every HOST_PLUGIN_HEARTBEAT_PERIOD. The signal is 'Healthy' when we have been able to
+        communicate with HostGAPlugin at least once in the last HOST_PLUGIN_HEALTH_PERIOD.
+        """
+
+        if last_host_plugin_heartbeat is None:
+            last_host_plugin_heartbeat = datetime.datetime.utcnow() - MonitorHandler.HOST_PLUGIN_HEARTBEAT_PERIOD
+
+        if datetime.datetime.utcnow() < (last_host_plugin_heartbeat + MonitorHandler.HOST_PLUGIN_HEARTBEAT_PERIOD):
+            return last_host_plugin_heartbeat
+
```

Comment:
This construct is used over and over again.  This should be refactored into something for better re-use.  Please open an issue if you cannot address it for this commit.

Reply:
I have made some changes here, which I will push. Let me know if you think it still requires further refactoring.